ambari-server/src/main/java/org/apache/ambari/server/events/listeners/alerts/AlertDefinitionsUIUpdateListener.java

```diff
@@ -137,13 +136,14 @@ public void onServiceComponentUninstalled(ServiceComponentUninstalledEvent event
 
   private void handleSingleDefinitionChange(AlertDefinitionEventType eventType, AlertDefinition alertDefinition) throws AmbariException {
     LOG.info("{} alert definition '{}'", eventType, alertDefinition);
-    Set<String> hosts = helper.get().invalidateHosts(alertDefinition);
-    for (String hostName : hosts) {
+    Cluster cluster = clusters.get().getCluster(alertDefinition.getClusterId());
+    helper.get().invalidateHosts(alertDefinition); // do we need to invalidate, what's the purpose of this?
+    for (String hostName : alertDefinition.matchingHosts(cluster)) {
```

Comment:
Since the definition is tied to a cluster, do you need to pass in the cluster instance?